src/camera.py

```diff
@@ -0,0 +1,181 @@
+
+
+    def pan(self, degree):
+        """
+        Pans the camera.
+
+        Controls the horizontal movement of the camera in [0, 180] degrees.
+        0 refers to the vehicle's right while 180 refers to the vehicle's left.
+        The degrees will be clamped to [0, 180].
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        self._current_pan_angle = np.clip(degree, self.MIN_DEGREE,
+                                                  self.MAX_PAN_DEGREE)
+        self._pan_servo.write(self._current_pan_angle + self._pan_offset)
+
+
+    def calibrate_pan(self, degree):
+        """
+        Calibrates the camera's panning servo.
+
+        Writes a new pan offset value to the config file and changes the current
+        pan offset. Calibration value can be negative.
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        self._pan_offset = int(np.clip(degree, self.MIN_PAN_CALI_DEGREE,
+                                               self.MAX_PAN_CALI_DEGREE))
+        self._db.set('pan_offset', self._pan_offset)
+
+
+    def get_pan_angle(self):
+        """
+        Gets the current pan angle of the camera.
+
+        :return: The current pan angle (in degrees).
+        :rtype: int
+        """
+        return self._current_pan_angle
+
+
+    def tilt(self, degree):
+        """
+        Tilts the camera.
+
+        Controls the vertical movement of the camera in [0, 150] degrees.
+        0 is directly in front of the car, 90 faces upwards, and 150 faces the
+        Raspberry Pi. The degrees will be clamped to [0, 150].
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        self._current_tilt_angle = np.clip(degree, self.MIN_DEGREE,
+                                                   self.MAX_TILT_DEGREE)
+        self._tilt_servo.write(self._current_tilt_angle + self._tilt_offset)
+
+
+    def calibrate_tilt(self, degree):
+        """
+        Calibrates the camera's tilting servo.
+
+        Writes a new tilt offset value to the config file and changes the current
+        tilt offset. Calibration value can be negative.
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        self._tilt_offset = int(np.clip(degree, self.MIN_TILT_CALI_DEGREE,
+                                                self.MAX_TILT_CALI_DEGREE))
+        self._db.set('tilt_offset', self._tilt_offset)
+
+
+    def get_tilt_angle(self):
+        """
+        Gets the current tilt angle of the camera.
+
+        :return: The current tilt angle (in degrees).
+        :rtype: int
+        """
+        return self._current_tilt_angle
+
+
+def main():
+    """
+    Demonstrates the camera's panning and tilting functionality.
+
+    Creates a camera object and pans from 0 to 180, then from 180 to 0. The
+    camera then tilts from 0 to 150, then 150 to 0.
+    """
+    sleep_time = 0.05
```

Comment:
The variable name should describe why we need to sleep. It's clear that it's a sleep time, but why is it necessary?